BookLibrary-cdi/src/main/java/org/booklibrary/app/persistence/session/common/AbstractGenericEntityPersistence.java

```diff
@@ -0,0 +1,127 @@
+package org.booklibrary.app.persistence.session.common;
+
+import org.booklibrary.app.persistence.entity.AbstractBaseEntity;
+import org.booklibrary.app.persistence.id.EntityIdentifier;
+import org.booklibrary.app.persistence.session.common.GenericFacadeLocal;
+import org.booklibrary.app.persistence.session.common.GenericHomeLocal;
+import org.slf4j.Logger;
+
+import javax.persistence.EntityManager;
+import java.lang.reflect.ParameterizedType;
+import java.util.List;
+
+/**
+ * Base abstract class for all entity persistence related classes.
+ *
+ * @see org.booklibrary.app.persistence.session.common.GenericHomeLocal
+ * @see org.booklibrary.app.persistence.session.common.GenericFacadeLocal
+ */
+public abstract class AbstractGenericEntityPersistence<T extends AbstractBaseEntity, PK>
+        implements GenericHomeLocal<T, PK>, GenericFacadeLocal<T, PK> {
+
+    private Class<T> entityClass;
+
+    // will be injected in implementations
+    public abstract EntityManager getEntityManager();
+    public abstract Logger getLogger();
+
+    public AbstractGenericEntityPersistence() {
+        this.entityClass = (Class<T>) ((ParameterizedType)
+                getClass().getGenericSuperclass()).getActualTypeArguments()[0];
+    }
+
+    @Override
+    public T findByPk(PK key) {
+
+        if (key == null) {
+            String errorMsg = "null argument passed to find";
+            getLogger().error(errorMsg);
+            throw new IllegalArgumentException(errorMsg);
+        }
+
+        getLogger().debug("find invoked for entity of type [" + entityClass.getCanonicalName() +
+                "] with pk [" + key + "]");
+
+        T entity = getEntityManager().find(entityClass, key);
+        getLogger().debug("result entity: {}", entity);
+        return entity;
+    }
+
+    @Override
+    public T findByUuid(String uuid) {
+        if (uuid == null) {
+            String errorMsg = "null argument passed to find";
+            getLogger().error(errorMsg);
+            throw new IllegalArgumentException(errorMsg);
+        }
+        PK key = (PK) new EntityIdentifier(uuid);
+        return findByPk(key);
+    }
+
+    @Override
+    public List<T> findAll() {
+        getLogger().debug("findAll invoked for entities of type: {}", entityClass.getCanonicalName());
+        List<T> resultList = getEntityManager().createQuery("FROM "
+                + entityClass.getName()).getResultList();
+        getLogger().debug("result list size: {}", resultList.size());
+        return resultList;
+    }
+
+    @Override
+    public T save(T obj) throws Exception {
```

Comment:
Do not throw generic Exception. Either create your own class that extends Exception or throw some existing, but specific exception.